### ISSUE TITLE
Writing to DB

### ISSUE THREAD

**Issue by panas89** [jtablesaw/tablesaw#529]
Is there a way to write a table to DB without explicitly creating the table in sql

**Comment by benmccann** [comment]
We only have a sql reader at the moment

**Comment by lwhite1** [comment]
@panas89 Just to make sure I understand, 

You're looking for a method that would write the contents of a Tablesaw table into a database table with the same structure, and that tablesaw should bundle the table creation and the insert operations in that method (assuming the table doesn't already exist). Is that right?

**Comment by panas89** [comment]
Yes exactly like you described. Maybe provide the option of replacing the table. Is an implementation that serves MYSQL, MSQL server, and postgres possible?

**Comment by albalkum** [comment]
I created some utilities to accomplish this functionality.  I'd be perfectly happy adapting them to fit into DataFrameWriter if that'd be appreciated.  @lwhite1 

If so, a question about implementation:
From seeing a comment in DataFrameWriter, it looks as if the goal is to move away from methods like csv() and utilize DataWriter by using to() methods.  Would a toConnection(java.sql.Connection connection) method be appropriate?  As DataWriter relies on Destination objects, Destination would need to be modified to accept a sql Connection object instead.

**Comment by benmccann** [comment]
`DataWriter` is implemented by every writing package (html, json, Excel, csv, etc.) It allows writing a binary output to streams, files, etc. since these are all file formats.

If `DataWriter` added a method to its interface for writing to a database these other writers would have to implement it. I'm not sure how you would write these formats to a database generally. Json may make sense for some databases like [postgres](https://www.postgresql.org/docs/9.3/functions-json.html), but in general I think you couldn't write most of these formats directly to a DB in the way that you could to a file or network destination, so you could probably ignore `DataWriter` for now.

**Comment by lwhite1** [comment]
@albalkum I was about to suggest you check with @benmccann , but he just weighed in, and his response is pretty much what i would have said. If i were to imagine a solution for writing to a database, there would be a SqlResultSetWriter analogous to the existing SqlResultSetReader, perhaps using the same mappings and so on. Is your solution along those lines?

**Comment by albalkum** [comment]
Thanks for the feedback!

Yes, essentially I have a writer that accepts a Connection, Table, and an options object (options for append/replace/create & row chunk size).  Based on the options, I create the queries and insert rows using PreparedStatements  I'll have to do some more extensive testing with the create option.  I've only ever used it with SQLite, which isn't very picky about types.

A quick question on code style: I pretty heavily use Streams (I have a Python & functional background) but should I refactor my code to use traditional for loops and such?  From my perusal of the codebase I get the sense that's the preferred style.

**Comment by lwhite1** [comment]
There’s no preference between for loops and streams.  Streams are fine.
